CheckIn/AttendanceReport.ascx.cs

```diff
@@ -0,0 +1,656 @@
+        }
+
+        /// <summary>
+        /// Handles the Click event of the btnRunReport control.
+        /// </summary>
+        /// <param name="sender">The source of the event.</param>
+        /// <param name="e">The <see cref="EventArgs"/> instance containing the event data.</param>
+        protected void btnRunReport_Click( object sender, EventArgs e )
+        {
+            GetAttendees();
+        }
+
+        #endregion
+
+        #region Methods
+
+        /// <summary>
+        /// Sets up everything.
+        /// </summary>
+        private void SetUpEverything()
+        {
+            SetCampusFilterVisibility();
+            LoadSchedulesDropdown();
+            LoadCheckInConfigurationGroupTypesDropdown();
+            LoadCheckInAreasDropdown();
+            GetAttendees();
+        }
+
+        /// <summary>
+        /// Sets the campus filter visibility.
+        /// </summary>
+        private void SetCampusFilterVisibility()
+        {
+            cpCampus.Visible = GetAttributeValue( AttributeKeys.ShowCampusFilter ).AsBoolean();
+
+            if ( cpCampus.Visible )
+            {
+                if ( GetAttributeValue( AttributeKeys.DefaultCampus ).IsNotNullOrWhiteSpace() )
+                {
+                    cpCampus.SelectedCampusId = CampusCache.Get( GetAttributeValue( AttributeKeys.DefaultCampus ) ).Id;
+                    SelectedCampus = cpCampus.SelectedValue;
+                }
+            }
+            else
+            {
+                cpCampus.SelectedValue = null;
+                SelectedCampus = null;
+                SetAttributeValue( AttributeKeys.DefaultCampus, null );
+            }
+        }
+
+        /// <summary>
+        /// Loads the schedules dropdown.
+        /// </summary>
+        private void LoadSchedulesDropdown()
+        {
+            ddlSchedule.DataSource = this.Schedules;
+            ddlSchedule.DataTextField = "Value";
+            ddlSchedule.DataValueField = "Key";
+            ddlSchedule.DataBind();
+        }
+
+        /// <summary>
+        /// Gets the schedules.
+        /// </summary>
+        /// <returns></returns>
+        private Dictionary<int, string> GetSchedules()
+        {
+            var categoryIds = new List<int>();
+            var schedules = new Dictionary<int, string>();
+
+            using ( var rockContext = new RockContext() )
+            {
+                var categoriesGuids = GetAttributeValues( AttributeKeys.ScheduleCategories ).AsGuidList();
+                if ( categoriesGuids.Count > 0 )
+                {
+                    categoryIds = new CategoryService( rockContext ).GetByGuids( categoriesGuids ).Select( c => c.Id ).ToList();
+                }
+
+                schedules = new ScheduleService( rockContext )
+                    .Queryable()
+                    .AsNoTracking()
+                    .Where( s => s.CategoryId.HasValue && categoryIds.Contains( s.CategoryId.Value ) )
```

Comment:
@treyhendon I tried to just run this without configuring any block settings and the schedule dropdown would never populate. It looks like you are checking if the schedule doesn't have a category but not if the categoryIds list/setting is blank. 

Just off the top of my head, not tested:
```suggestion
                    .Where( s => s.CategoryId.HasValue && ( categoryIds.Contains( s.CategoryId.Value ) || !categoryIds.Any() ) )
```

Reply:
Great catch!

Reply:
Resolved with: https://github.com/KingdomFirst/RockBlocks/pull/35/commits/80520da62b70a9e12eb4398760f1fdda244123a7